### PR TITLE
Added Bluesky icon to the Social Icon Block

### LINKED ISSUE #61325: Social Icons: Add Bluesky social icon
## What problem does this address?

We'd love to have Bluesky's social icon available.

image

## What is your proposed solution?

- Add the Bluesky social icon [as we did for WhatsApp](https://github.com/WordPress/gutenberg/pull/42137).
- Be aware of its usage with different colors [as pointed here](https://github.com/WordPress/gutenberg/pull/42137#issuecomment-1230084478).

### PR DESCRIPTION
**Pull request by amitraj2203**
## What?
I have added the Bluesky icon to the Social Icons Block.

## Why?
Bluesky has just been added as a trusted oEmbed provider in core.

This is related to: #61325 

## Testing Instructions
1. Open a Post or Page
2. Add a Social Block
4. Add the Bluesky Icon
4. Add content to it, normally https://bsky.app/profile/your-username
5. Save and make sure it appears as expected in the editor as well as the preview

## Screenshots or screencast 

In the editor:
image

Result in the preview:
image

### CODE REVIEW

**Comment by amitraj2203** [comment]
Hi @t-hamano and @talldan 
As the [issue](https://github.com/WordPress/gutenberg/issues/61325) for this PR has been marked closed, should I close this PR or keep it open?

**Comment by talldan** [comment]
@amitraj2203 It looks like there may have been a mix-up that resulted in that issue being closed. I've re-opened it.

File: packages/block-library/src/social-link/icons/bluesky.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * WordPress dependencies
+ */
+import { Path, SVG } from '@wordpress/primitives';
+
+export const BlueskyIcon = () => (
+	<SVG width="24" height="24" viewBox="0 0 320 256" version="1.1">
+		<Path d="M69.364 19.146c36.687 27.806 76.147 84.186 90.636 114.439 14.489-30.253 53.948-86.633 90.636-114.439C277.107-.917 320-16.44 320 32.957c0 9.865-5.603 82.875-8.889 94.729-11.423 41.208-53.045 51.719-90.071 45.357 64.719 11.12 81.182 47.953 45.627 84.785-80 82.874-106.667-44.333-106.667-44.333s-26.667 127.207-106.667 44.333c-35.555-36.832-19.092-73.665 45.627-84.785-37.026 6.362-78.648-4.149-90.071-45.357C5.603 115.832 0 42.822 0 32.957 0-16.44 42.893-.917 69.364 19.147Z" />
+	</SVG>
```
Comment (t-hamano) [inline packages/block-library/src/social-link/icons/bluesky.js]:
```suggestion
	<SVG width="24" height="24" viewBox="0 0 24 24" version="1.1">
		<Path d="M6.3,4.2c2.3,1.7,4.8,5.3,5.7,7.2.9-1.9,3.4-5.4,5.7-7.2,1.7-1.3,4.3-2.2,4.3.9s-.4,5.2-.6,5.9c-.7,2.6-3.3,3.2-5.6,2.8,4,.7,5.1,3,2.9,5.3-5,5.2-6.7-2.8-6.7-2.8,0,0-1.7,8-6.7,2.8-2.2-2.3-1.2-4.6,2.9-5.3-2.3.4-4.9-.3-5.6-2.8-.2-.7-.6-5.3-.6-5.9,0-3.1,2.7-2.1,4.3-.9h0Z" />
	</SVG>
```

File: packages/block-library/src/social-link/index.php [hunk 2]
```diff
@@ -153,6 +153,10 @@ function block_core_social_link_services( $service = '', $field = '' ) {
...
 		),
+		'bluesky'       => array(
+			'name' => 'Bluesky',
+			'icon' => '<svg width="24" height="24" viewBox="0 0 320 256" version="1.1" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false"><path d="M69.364 19.146c36.687 27.806 76.147 84.186 90.636 114.439 14.489-30.253 53.948-86.633 90.636-114.439C277.107-.917 320-16.44 320 32.957c0 9.865-5.603 82.875-8.889 94.729-11.423 41.208-53.045 51.719-90.071 45.357 64.719 11.12 81.182 47.953 45.627 84.785-80 82.874-106.667-44.333-106.667-44.333s-26.667 127.207-106.667 44.333c-35.555-36.832-19.092-73.665 45.627-84.785-37.026 6.362-78.648-4.149-90.071-45.357C5.603 115.832 0 42.822 0 32.957 0-16.44 42.893-.917 69.364 19.147Z"></path></svg>',
```
Comment (t-hamano) [inline packages/block-library/src/social-link/index.php]:
```suggestion
			'icon' => '<svg width="24" height="24" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false"><path d="M6.3,4.2c2.3,1.7,4.8,5.3,5.7,7.2.9-1.9,3.4-5.4,5.7-7.2,1.7-1.3,4.3-2.2,4.3.9s-.4,5.2-.6,5.9c-.7,2.6-3.3,3.2-5.6,2.8,4,.7,5.1,3,2.9,5.3-5,5.2-6.7-2.8-6.7-2.8,0,0-1.7,8-6.7,2.8-2.2-2.3-1.2-4.6,2.9-5.3-2.3.4-4.9-.3-5.6-2.8-.2-.7-.6-5.3-.6-5.9,0-3.1,2.7-2.1,4.3-.9h0Z"></path></svg>',
```

**Review by t-hamano — COMMENTED** [review]
@amitraj2203 Thanks for the PR! Sorry for closing the issue by mistake 😅

I checked this PR and this icon looks slightly larger than the others.

image

Also, the size of the view box will need to be unified to 24.

Fortunately, there is a normalized icon added in #61352, so it's a good idea to use it. That's why I left the icon change as a comment.

Additionally, if we add new social icons, we also need to [add fixtures](https://github.com/WordPress/gutenberg/blob/trunk/test/integration/fixtures/blocks/README.md#creating-fixtures) at the same time.

You should be able to create it by following the steps below.

- Create a `/test/integration/fixtures/blocks/core__social-link-bluesky.html` file and write the following:
  ```html
  <!-- wp:social-link-bluesky {"url":"https://example.com/"} /-->
  ```
- Run `npm run fixtures:regenerate`.
- Commit the generated files.

**Comment by amitraj2203** [comment]
Hi @t-hamano, I have made the changes as per your suggestions. Please have a look at it.

Thanks!

**Review by t-hamano — APPROVED** [review]
LGTM 👍

image

**Pull request merged**